hub/util/diff.py

```diff
@@ -281,6 +343,29 @@ def filter_data_updated(changes: Dict[str, Dict]):
         change["data_updated"] = upd
 
 
+def filter_deleted_no_rename(dataset_changes, tensor_changes):
+    """Remove deleted tensors and tensors renamed to same name from diff"""
+    rm = []
+    renamed = dataset_changes.get("renamed")
+    deleted = dataset_changes.get("deleted")
+    if renamed:
+        for old_name, new_name in renamed.items():
+            if old_name == new_name:
```

Comment:
when would old_name be same as new_name?

Reply:
you can rename a tensor multiple times through commits, it could have the same name in two commits. There should be no rename diff between those two commits.